pkg/converter/converter_test.go

```diff
@@ -0,0 +1,176 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package converter
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/istio-ecosystem/sail-operator/pkg/test/project"
+	"github.com/istio-ecosystem/sail-operator/pkg/test/util/supportedversion"
+	"github.com/istio-ecosystem/sail-operator/tests/e2e/util/shell"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	controlPlaneNamespace = "istio-system"
+	// Test converter with latest istio version
+	istioVersion = supportedversion.List[len(supportedversion.List)-1].Name
+)
+
+func TestSimpleYamlConversion(t *testing.T) {
+	g := NewWithT(t)
+
+	istioYamlText := `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:`
+
+	sailYamlText := `apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s`
+	sailYamlText = fmt.Sprintf(sailYamlText, istioVersion, controlPlaneNamespace)
+
+	istioYamlFileWithPath, err := saveYamlToFile(istioYamlText)
+	g.Expect(err).To(Succeed(), "failed to write YAML file")
+
+	g.Expect(executeConfigConverter(istioYamlFileWithPath, controlPlaneNamespace, istioVersion)).To(Succeed(),
+		"error in execution of ./configuration-converter.sh")
+	isConversionValidated, err := validateYamlContent(sailYamlText)
+	g.Expect(err).To(Succeed(), "Can not open file to compare")
+	g.Expect(isConversionValidated).To(BeTrue(), "Converted content is not as expected")
+
+	err = os.Remove(filepath.Join(project.RootDir, "tools", "istioConfig.yaml"))
+	g.Expect(err).To(Succeed(), "Unable to delete istioConfig.yaml")
+	err = os.Remove(filepath.Join(project.RootDir, "tools", "sail-operator-config.yaml"))
+	g.Expect(err).To(Succeed(), "Unable to delete sail-operator-config.yaml")
+}
+
+func TestComplexYamlConversion(t *testing.T) {
```

Comment:
Since this and the previous test both test the same function, it would be better if both tests were implemented in the same test function, with table-based sub-tests. Something like:
```
func TestConversion(t *testing.T) {
    var flagtests = []struct {
        name  string
        input string
        expectedOutput string
    }{
        {"simple", "simple input yaml", "simple output yaml"},
        {"complex", "complex input yaml", "complex output yaml"},
    }    

    for _, tt := range testcases {
        t.Run(tt.name, func(t *testing.T) {
            ...
        })
    }
}
```

This would reduce code duplication and allow the next person to easily add another sub-test.